core/src/main/java/nl/nn/adapterframework/senders/ImapSender.java

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright 2020 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package nl.nn.adapterframework.senders;
+
+import javax.mail.Message;
+
+import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.filesystem.FileSystemSender;
+import nl.nn.adapterframework.filesystem.ImapFileSystem;
+
+public class ImapSender extends FileSystemSender<Message, ImapFileSystem> {
+	public final String IMAP_FILE_SYSTEM ="nl.nn.adapterframework.filesystem.ImapFileSystem";
```

Comment:
Ik zie dit vaker, is het niet mogelijk in `ImapFileSystem` een `public final String ImapFileSystem = this.getClass().getName()` te doen? Mocht dan de naam veranderen hoef je niet overal de referenties aan te passen...

Reply:
Nee, dat kan niet. Dan moppert IbisDocRef: The value for annotation attribute IbisDocRef.value must be a constant expression